src/components/App.js

```diff
@@ -1,113 +1,211 @@
-          return response.json();
-        }
-        throw Error(`Request rejected with status ${response.status}`);
-      })
-      .then((userData) => {
-        console.log(userData);
-        const user = userData.display_name;
-        this.setState(prevState => ({
-          ...prevState,
-          user,
-          error: false,
+  getProfile = async (token) => {
+    try {
+      const userProfile = await fetchProfile(token);
+      this.setState({
+        user: {
+          name: userProfile.display_name,
+          id: userProfile.id,
           errorMessage: '',
-        }));
-      })
-      .catch((error) => {
-        this.setState({
-          error: true,
+        },
+      });
+    } catch (error) {
+      this.setState({
+        user: {
           errorMessage: error.message,
-        });
+        },
+      });
+    }
+  }
+
+  getUserPlaylists = async (token, endpoint = 'https://api.spotify.com/v1/me/playlists') => {
+    const { needsRefresh } = this.state.userPlaylists;
+    try {
+      const playlistsObject = await fetchSomePlaylists(token, endpoint);
+      this.setState((prevState) => {
+        const existingItems = needsRefresh
+          ? []
+          : prevState.userPlaylists.items;
+        return {
+          userPlaylists: {
+            items: [...existingItems, ...playlistsObject.items],
+            nextPlaylistsEndpoint: playlistsObject.next,
+            errorMessage: '',
+            needsRefresh: false,
+          },
+        };
       });
+    } catch (error) {
+      this.setState(prevState => ({
+        userPlaylists: {
+          ...prevState.userPlaylists,
+          errorMessage: error.message,
+        },
+      }));
+    }
+  }
+
+  sortPlaylists = (sortBy, sortDescending = false) => {
+    const { items } = this.state.userPlaylists;
+    const playlistsMap = items.map((playlist, i) => ({
+      index: i,
+      name: playlist.name.toLowerCase(),
+      total: playlist.tracks.total,
+    }));
+    playlistsMap.sort((a, b) => {
+      if (a[`${sortBy}`] > b[`${sortBy}`]) {
+        return 1;
+      }
+      if (a[`${sortBy}`] < b[`${sortBy}`]) {
+        return -1;
+      }
+      return 0;
+    });
+    if (sortDescending) {
+      playlistsMap.reverse();
+    }
+    const playlistsSorted = playlistsMap.map(mapItem => items[mapItem.index]);
+    this.setState(prevstate => ({
+      userPlaylists: {
+        ...prevstate.userPlaylists,
+        items: [...playlistsSorted],
+      },
+    }));
+  };
+
+  refreshPlaylists = () => {
+    this.setState(prevstate => ({
+      userPlaylists: {
+        ...prevstate.userPlaylists,
+        needsRefresh: true,
+      },
+    }));
+  }
+
+  setCurrentPlaylist = (playlist) => {
+    const { currentPlaylist } = this.state;
+    const selectedPlaylist = { ...playlist };
+    if (selectedPlaylist.id !== currentPlaylist.id) {
+      this.setState({ currentPlaylist: selectedPlaylist });
+    }
+  };
+
+  updateUserPlaylists = (playlist) => {

```

Comment:
Will either add a new playlist to the list or update an existing playlists' details, based on whether the playlist already exists in the list.